pkg/kubelet/kubelet.go

```diff
@@ -1715,6 +1721,12 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	// Fetch the pull secrets for the pod
 	pullSecrets := kl.getPullSecretsForPod(pod)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		if !kl.podIsTerminated(pod) && !kubepod.IsStaticPod(pod) {
```

Comment:
We do it here because this conveniently takes care of retries in cases where resize is possible, but not right now. (the [Deferred](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#resize-status) case)

Since we need to retry, we can't lose the mutex, can we?

Reply:
Sigh, I don't like this kind of mutex... But that makes sense to me. :/

Reply:
So, I initially preferred the idea of [retrying pending pod resizes](https://github.com/kubernetes/enhancements/pull/686#issuecomment-490629494) when some other non-BestEffort pods depart. (event based vs periodic) but got significant pushback since this deviates from normal K8s model or retries.

Do you think it is worth revisiting that approach? I think it can help get rid of the lock (I'll have to look closely to be sure) if in HandlePodDelete, we gathered a list of active pods whose PodStatus.Resize == Deferred, perhaps sort them by class/priority/FCFS etc , and retry the resize for those pods. This would need some bake time but if it works reliably, we can toss out the periodic retries and the lock.

Reply:
Having a deferred list of pod to update, and re-admit periodically in the main syncloop seems better to me, that won't have race condition with pod worker.

Reply:
I think we can actually just let syncloop periodically scan the pod manager, and admit pod that `request != allocated` periodically, and admit some of them when feasible. Will that work?

That seems much better than letting pod worker does it.

Reply:
It might, I have to think through this. I like event-triggered because it is more deterministic. We are attempting something when we know there might be a shot and minimize futile retries that we may have with periodic-only retry approach. I'll take a fix-before-beta issue for this as well.

Reply:
I strongly prefer the `HandlePodUpdates` + `SyncLoop` model, it is the right way of doing this within kubelet.

Please add a TODO, and be sure to discuss and address this before beta.

Reply:
Sure. I've added a TODO in the code and will merge it in next commit.
I'm tracking this with other outstanding beta issues on the KEP thread https://github.com/kubernetes/enhancements/issues/1287#issue-503823628